commentingapp/commenting_new.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+
+import requests
+import collections
+import os
+from urlparse import urljoin
+
+
+
+class DisqusCommenting(object):
+    """ A class for getting disqus comments per url, also features getting flagged comments."""
+  
+    base_disqus_url  = "http://disqus.com/api/"
+
+
+    def check_internet_connection(self):
+        """ Checks for the internet connection."""
+  
+        try:
+            requests.get(self.base_disqus_url, timeout = 10)
+            self.internet_status = {"status":True, "message": "Connection Passed."}
+
+        except (requests.exceptions.Timeout, requests.exceptions.ConnectionError):
+            self.internet_status = {"status": False, "message": "Please Check the internet Connection."}
+
+        return self.internet_status["message"]
+
+    def check_authentication(self, public_key, forum_name, api_version=3.0):
+
+        """ Checks if public key and forum is valid. Returns the public key, forum name for Disqus API."""
+        # @TODO - Optional Authentication for read/write/moderate.
```

Comment:
Not yet. That's the one bit I am working upon. I am actually thinking of giving the admin the right to delete the comments from the disqus database, from the admin-commenting page itself.